### PR TITLE
uutils-coreutils 0.0.1 (new formula)

### PR DESCRIPTION
**Pull request by vladimyr**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

**NOTE:** Version number indicates an early stage of development while in reality project is a continuing 6-year long journey.

### CODE REVIEW

File: Formula/uutils.rb [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  depends_on "rust" => :build
+  depends_on "sphinx-doc" => :build
+
+  conflicts_with "coreutils", :because => "uutils and coreutils install the same binaries"
+  conflicts_with "aardvark_shell_utils", :because => "both install `realpath` binaries"
+  conflicts_with "truncate", :because => "both install `truncate` binaries"
+
+  def install
+    man1.mkpath
+    ENV["PATH"]="#{Formula["make"].opt_libexec}/gnubin:#{ENV["PATH"]}"
```
Comment (vladimyr) [inline Formula/uutils.rb:18]: I'm not `make` expert by any means but I believe https://github.com/uutils/coreutils/pull/1486 could make this replaceable by simple `gmake` invocation.

File: Formula/uutils.rb [hunk 2]
```diff
@@ -0,0 +1,70 @@
+class Uutils < Formula
```
Comment (MikeMcQuaid) [inline Formula/uutils.rb:1]: `UutilsCoreutils` would be more appropriate I think

File: Formula/uutils.rb [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  def caveats
+    <<~EOS
+      Commands also provided by macOS have been installed with the prefix "u".
+      If you need to use these commands with their normal names, you
+      can add a "uubin" directory to your PATH from your bashrc like:
+        PATH="#{opt_libexec}/uubin:$PATH"
+    EOS
+  end
+
+  def coreutils_filenames(dir)
```
Comment (MikeMcQuaid) [inline Formula/uutils.rb:54]: Make this a variable rather than a function

**Comment by MikeMcQuaid** [comment]
I'm game for this being included once the formula is cleaned up a bit.

**Comment by vladimyr** [comment]
> I'm game for this being included once the formula is cleaned up a bit.

Glad to hear that! Fixing it right away.

File: Formula/uutils.rb (hunk 3 above)
Comment (vladimyr) [reply Formula/uutils.rb:54]: It is actually an exact copy of https://github.com/Homebrew/homebrew-core/blob/39df0a6f/Formula/coreutils.rb#L79-L87 Are you sure about this one? 🤔

Comment (MikeMcQuaid) [reply Formula/uutils.rb]: I missed that it was taking an argument so: yeh, this is fine 👍

File: Formula/uutils-coreutils.rb [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  depends_on "sphinx-doc" => :build
+
+  conflicts_with "coreutils", :because => "uutils-coreutils and coreutils install the same binaries"
+  conflicts_with "aardvark_shell_utils", :because => "both install `realpath` binaries"
+  conflicts_with "truncate", :because => "both install `truncate` binaries"
+
+  def install
+    man1.mkpath
+    ENV["PATH"]="#{Formula["make"].opt_libexec}/gnubin:#{ENV["PATH"]}"
+    system "make", "install",
```
Comment (MikeMcQuaid) [inline Formula/uutils-coreutils.rb:19]:
```suggestion
    man1.mkpath
    
    ENV.prepend_path Formula["make"].opt_libexec/"gnubin"
    
    system "make", "install",
```

File: Formula/uutils-coreutils.rb [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+           "PROG_PREFIX=u",
+           "PREFIX=#{prefix}",
+           "SPHINXBUILD=#{Formula["sphinx-doc"].opt_bin}/sphinx-build"
+
+    # Symlink all commands into libexec/uubin without the 'u' prefix
+    coreutils_filenames(bin).each do |cmd|
+      (libexec/"uubin").install_symlink bin/"u#{cmd}" => cmd
+    end
+    # Symlink all man(1) pages into libexec/uuman without the 'u' prefix
```
Comment (MikeMcQuaid) [inline Formula/uutils-coreutils.rb:28]:
```suggestion
    end
    
    # Symlink all man(1) pages into libexec/uuman without the 'u' prefix
```

File: Formula/uutils-coreutils.rb [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    coreutils_filenames(man1).each do |cmd|
+      (libexec/"uuman"/"man1").install_symlink man1/"u#{cmd}" => cmd
+    end
+    libexec.install_symlink "uuman" => "man"
```
Comment (MikeMcQuaid) [inline Formula/uutils-coreutils.rb:32]:
```suggestion
    end
    
    libexec.install_symlink "uuman" => "man"
```

File: Formula/uutils-coreutils.rb [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+    # Symlink non-conflicting binaries
+    no_conflict = %w[
+      base32 dircolors factor hashsum hostid nproc numfmt pinky ptx realpath
+      shred shuf stdbuf tac timeout truncate
+    ]
+    no_conflict.each do |cmd|
```
Comment (MikeMcQuaid) [inline Formula/uutils-coreutils.rb:39]:
```suggestion
    %w[
      base32 dircolors factor hashsum hostid nproc numfmt pinky ptx realpath
      shred shuf stdbuf tac timeout truncate
    ].each do |cmd|
```

File: Formula/uutils-coreutils.rb [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+  depends_on "sphinx-doc" => :build
+
+  conflicts_with "coreutils", :because => "uutils-coreutils and coreutils install the same binaries"
+  conflicts_with "aardvark_shell_utils", :because => "both install `realpath` binaries"
+  conflicts_with "truncate", :because => "both install `truncate` binaries"
+
+  def install
+    man1.mkpath
+
+    ENV.prepend_path Formula["make"].opt_libexec/"gnubin"
```
Comment (MikeMcQuaid) [inline Formula/uutils-coreutils.rb:19]:
```suggestion
    ENV.prepend_path "PATH", Formula["make"].opt_libexec/"gnubin"
```

sorry, my bad

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks so much for your contribution! Without people like you submitting PRs we couldn't run this project. You rock, @vladimyr!

**Pull request closed without merging**